lib/ansible/modules/storage/netapp/na_cdot_aggregate.py

```diff
@@ -0,0 +1,317 @@
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+DOCUMENTATION = '''
+
+module: na_cdot_aggregate
+
+short_description: Manage NetApp cDOT aggregates
+version_added: '2.3'
+author: Sumit Kumar (sumit4@netapp.com)
+
+description:
+- Create or destroy aggregates on NetApp cDOT
+
+options:
+
+  state:
+    required: true
+    description:
+    - Whether the specified aggregate should exist or not.
+    choices: ['present', 'absent']
+
+  name:
+    required: true
+    description:
+    - The name of the aggregate to manage
+
+  new_name:
+    required: false
+    description:
+    - Rename the aggregate
+
+  disk_count:
+    required: false
+    notes: required when state == 'present'
+    description:
+    - Number of disks to place into the aggregate, including parity disks. The disks in this newly-created aggregate come from the spare disk pool. The smallest disks in this pool join the aggregate first, unless the "disk-size" argument is provided. Either "disk-count" or "disks" must be supplied. Range [0..2^31-1].
+
+  hostname:
+    required: true
+    description:
+    - hostname
+
+  username:
+    required: true
+    description:
+    - username
+
+  password:
+    required: true
+    description:
+    - password
+
+'''
+
+EXAMPLES = """
+    - name: Manage Aggregates
+          na_cdot_aggregate:
+            state: present
+            name: ansibleAggr
+            disk_count: 1
+            hostname: "{{ netapp_hostname }}"
+            username: "{{ netapp_username }}"
+            password: "{{ netapp_password }}"
+
+    - name: Manage Aggregates
+          na_cdot_aggregate:
+            state: present
+            name: ansibleAggr
+            new_name: ansibleAggrRenamed
+            hostname: "{{ netapp_hostname }}"
+            username: "{{ netapp_username }}"
+            password: "{{ netapp_password }}"
+
+"""
+
+RETURN = """
+msg:
+    description: Successfully created new aggregate (if not already available)
+    returned: success
+    type: string
+    sample: '{"changed": true}'
+
+msg:
+    description: Successfully renamed aggregate
+    returned: success
+    type: string
+    sample: '{"changed": true}'
+
+"""
+
+import sys
+import json
+import logging
+from itertools import ifilter
+from traceback import format_exc
+
+from ansible.module_utils.basic import *
+from ansible.module_utils.urls import *
+from ansible.module_utils.pycompat24 import get_exception
+
+import socket
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+
+from netapp_lib.api.zapi import zapi
```

Comment:
Import custom packages in try/except and handled with fail_json() in main() e.g.::

    try:
        import foo
        HAS_LIB=True
    except:
        HAS_LIB=False